### PR TITLE
[CBR-533] add a new endpoint to calculate the walletid for a given mnemonic

### PR DESCRIPTION
**Pull request by cleverca22**
## Type of change

- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🛠 New feature (non-breaking change which adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🏭 Refactoring that does not change existing functionality but does improve things like code readability, structure etc
- [ ] 🔨 New or improved tests for existing code
- [ ] ⛑ git-flow chore (backport, hotfix, etc)

## Developer checklist

- [ ] I have read the [style guide](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/style-guide.md) document, and my code follows the code style of this project.
- [ ] If my code deals with exceptions, it follows the [guidelines](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/exceptions.md).
- [ ] I have updated any documentation accordingly, if needed. Documentation changes can be reflected in opening a PR on [cardanodocs.com](https://github.com/input-output-hk/cardanodocs.com), amending the inline [Haddock](https://www.haskell.org/haddock/) comments, any relevant README file or one of the document listed in the [docs](https://github.com/input-output-hk/cardano-sl/tree/develop/docs) directory.
- [ ] CHANGELOG entry has been added and is linked to the correct PR on GitHub.

## Testing checklist

- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

## How to merge

Send the message `bors r+` to merge this PR. For more information, see
[`docs/how-to/bors.md`](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/how-to/bors.md).

### CODE REVIEW

**Comment by tatyanavych** [comment]
@cleverca22 Please, add CHANGELOG.md entry under `Unreleased` in `Features` section (the section does not exist - needs to be created). Could you also add `Fixes` section below for the existed entry?
So to have:

**Unreleased**

**Features**
* A new endpoint... (CBR-533 #<PR#>)

**Fixes**
* Correct change category...

File: wallet/src/Cardano/Wallet/API/Internal.hs [hunk 1]
```diff
@@ -3,13 +3,15 @@
 -- Daedalus client, and aren't useful for wallets, exchanges, and other users.
 module Cardano.Wallet.API.Internal where
 
+import Prelude
```
Comment (johnalotoski) [inline wallet/src/Cardano/Wallet/API/Internal.hs]: Indentation to match import lines below?  Also isn't this import already implicit?

**Comment by cleverca22** [comment]
@KtorZ without some chain state to fire up the wallet, i don't know how to test this endpoint fully in an automated manner

all other review comments should be good

**Comment by KtorZ** [comment]
@cleverca22 For the testing, have a look at some integration test scenarios (e.g. hhttps://github.com/input-output-hk/cardano-sl/blob/develop/wallet/test/integration/Test/Integration/Scenario/Wallets.hs#L579-L601) 

You should be able to do something like:

```hs
    scenario "Can query wallet id and balance using only its mnemonic" $
            fixture <- setup $ defaultSetup
                & initialCoins .~ [14]

            successfulRequest $ Client.deleteWallet
                $- (fixture ^. wallet . walletId)

            response <- request $ Client.calculateMnemonic 
                $- (fixture ^. backupPhrase)
                $- (Just True) 

            verify response
                [ expectFieldEqual walletId (fixture ^. wallet . walletId)
                , expectFieldEqual maybeBalance (Just 14)
                ]
```

+/- the compilation errors and testing with various scenarios (with the query parameter, without, testing on a wallet with no funds etc ..)

**Pull request merged**